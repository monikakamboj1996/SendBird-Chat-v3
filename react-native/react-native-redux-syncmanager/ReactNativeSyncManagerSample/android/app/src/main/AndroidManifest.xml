<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.reactnativesyncmanagersample">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <permission
        android:name="${applicationId}.permission.C2D_MESSAGE"
        android:protectionLevel="signature" />
    <uses-permission android:name="${applicationId}.permission.C2D_MESSAGE" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_icon"
        android:resource="@mipmap/ic_notification" />
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_channel_id"
        android:value="@string/default_notification_channel_id"/>
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:launchMode="singleTop"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

      <service android:name="io.invertase.firebase.messaging.RNFirebaseMessagingService">
        <intent-filter>
          <action android:name="com.google.firebase.MESSAGING_EVENT" />
        </intent-filter>
      </service>
      <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
    </application>

</manifest>
