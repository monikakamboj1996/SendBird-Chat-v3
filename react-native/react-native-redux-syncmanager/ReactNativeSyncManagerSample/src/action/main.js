
const Action = {
	Type: {
		CHANNEL_ADD: 'channel-add',
		CHANNEL_UPDATE: 'channel-update',
		CHANNEL_REMOVE: 'channel-remove',
		CHANNEL_MOVE: 'channel-move',
		CHANNEL_CLEAR: 'channel-clear'
	}
};
Action.addChannels = channels => {
	return {
		type: Action.Type.CHANNEL_ADD,
		payload: {
			channels
		}
	};
};
Action.updateChannels = channels => {
	return {
		type: Action.Type.CHANNEL_UPDATE,
		payload: {
			channels
		}
	};
}
Action.removeChannels = channels => {
	return {
		type: Action.Type.CHANNEL_REMOVE,
		payload: {
			channels
		}
	};
};
Action.moveChannels = channels => {
	return {
		type: Action.Type.CHANNEL_MOVE,
		payload: {
			channels
		}
	}
};
Action.clearChannels = () => {
	return {
		type: Action.Type.CHANNEL_CLEAR
	}
};

export default Action;