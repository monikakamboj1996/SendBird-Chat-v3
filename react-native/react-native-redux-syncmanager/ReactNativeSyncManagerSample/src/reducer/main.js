import Action from '../action/main';
import { findChannelIndex, getChannelIndex } from '../utils';

const _initialState = {
  channels: []
};

export const reducer = (state = _initialState, action) => {
  const newState = { ...state };
  switch (action.type) {
    case Action.Type.CHANNEL_ADD: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = findChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 0, channel);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_UPDATE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels[index] = channel;
        }
      }
      break;
    }
    case Action.Type.CHANNEL_REMOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const index = getChannelIndex(channel, newState.channels);
        if (index >= 0) {
          newState.channels.splice(index, 1);
        }
      }
      break;
    }
    case Action.Type.CHANNEL_MOVE: {
      newState.channels = [...newState.channels];
      for (let i = 0; i < action.payload.channels.length; i++) {
        const channel = action.payload.channels[i];
        const from = getChannelIndex(channel, newState.channels);
        if (from >= 0) {
          newState.channels.splice(from, 1);
        }
        const to = findChannelIndex(channel, newState.channels);
        newState.channels.splice(to, 0, channel);
      }
      break;
    }
    case Action.Type.CHANNEL_CLEAR: {
      newState.channels = [];
      break;
    }
  }
  return newState;
};
