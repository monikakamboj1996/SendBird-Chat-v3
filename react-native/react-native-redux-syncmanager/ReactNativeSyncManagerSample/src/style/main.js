
import { StyleSheet } from 'react-native';

export const style = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: '#f3f3f3'
	},
	actionButtonShadow: {
		shadowColor: "#333",
		shadowOpacity: 0.6,
		shadowOffset: {
			height: 1,
			width: 1
		}
	},
	actionButtonIcon: {
		fontSize: 20,
    height: 22,
    color: 'white'
	},
	signoutButtonContainer: {
		marginLeft: 0,
		marginRight: 10
	},
	signoutButton: {
		backgroundColor: 'transparent',
		paddingHorizontal: 10
	}
});