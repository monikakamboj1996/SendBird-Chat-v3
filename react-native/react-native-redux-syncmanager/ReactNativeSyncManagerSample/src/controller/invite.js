import React from 'react';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/Ionicons';
import { FlatList } from 'react-native-gesture-handler';

import SendBird from 'sendbird';

import { style } from '../style/invite';
import Action from '../action/invite';
import UserView from '../view/user';
import { toast } from '../utils';

let _selectedUserIds = [];

export class InviteController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    const channel = navigation.getParam('channel', null);
    return {
      title: !channel ? 'Create new channel' : 'Invite',
      headerRight: (
        <View style={{ flexDirection: 'row' }}>
          <Button
            containerStyle={style.headerInviteContainer}
            buttonStyle={style.headerInvite}
            icon={<Icon name="md-add" color="#7d62d9" size={28}></Icon>}
            onPress={() => {
              if (!channel) {
                if (_selectedUserIds.length > 0) {
                  const sb = SendBird.getInstance();
                  const params = new sb.GroupChannelParams();
                  params.addUserIds(_selectedUserIds);
                  sb.GroupChannel.createChannel(params, (channel, err) => {
                    if (!err) {
                      navigation.replace('Chat', { channel });
                    } else {
                      toast('Failed to create channel.');
                    }
                  });
                } else {
                  toast('Choose at least 1 user to invite.');
                }
              } else {
                channel.inviteWithUserIds(_selectedUserIds, (res, err) => {
                  if (!err) {
                    navigation.pop();
                  } else {
                    toast('Failed to invite users.');
                  }
                  isLoading = false;
                });
              }
            }}
          />
        </View>
      )
    };
  };
  constructor(props) {
    super(props);
    const sb = SendBird.getInstance();
    this.query = sb.createUserListQuery();
    this.channel = this.props.navigation.getParam('channel', null);
    this.state = {
      users: []
    };
  }
  static getDerivedStateFromProps(props, state) {
    state.users = props.users;
    return state;
  }
  componentDidMount() {
    this.next();
  }
  componentWillUnmount() {
    _selectedUserIds = [];
  }
  toggle(selected, user) {
    if (selected) {
      if (_selectedUserIds.indexOf(user.userId) < 0) {
        _selectedUserIds.push(user.userId);
      }
    } else {
      const index = _selectedUserIds.indexOf(user.userId);
      if (index >= 0) {
        _selectedUserIds.splice(index, 1);
      }
    }
  }
  next() {
    if (this.query.hasNext) {
      this.query.next((users, err) => {
        if (!err) {
          let filteredUsers = [];
          if (this.channel) {
            for (let i in users) {
              if (this.channel.members.map(m => m.userId).indexOf(users[i].userId) < 0) {
                filteredUsers.push(users[i]);
              }
            }
          } else {
            filteredUsers = users;
          }
          if (filteredUsers.length < this.query.limit && this.query.hasNext) {
            setTimeout(() => this.next(), 200);
          } else {
            this.props.fetchUsers(filteredUsers);
          }
        }
      });
    }
  }
  render() {
    return (
      <View>
        <FlatList
          data={this.state.users}
          keyExtractor={user => user.userId + ''}
          onEndReached={this.next.bind(this)}
          renderItem={bundle => (
            <UserView
              user={bundle.item}
              selectable={true}
              selected={_selectedUserIds.indexOf(bundle.item.userId) >= 0}
              onPress={this.toggle.bind(this)}
            />
          )}
        />
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { invite } = previousState;
    return { ...invite };
  },
  Action
)(InviteController);
