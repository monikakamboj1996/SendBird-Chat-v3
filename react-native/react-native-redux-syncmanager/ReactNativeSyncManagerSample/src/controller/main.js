import React from 'react';
import { Alert, Platform, View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';

import AsyncStorage from '@react-native-community/async-storage';
import { FlatList } from 'react-native-gesture-handler';
import ActionButton from 'react-native-action-button';
import Icon from 'react-native-vector-icons/Ionicons';
import firebase from 'react-native-firebase';

import SendBird from 'sendbird';
import SendBirdSyncManager from 'sendbird-syncmanager';

import { style } from '../style/main';
import Action from '../action/main';
import GroupChannelView from '../view/groupChannel';
import { navigator } from '../navigator';

export class MainController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    return {
      title: 'SendBird',
      headerRight: (
        <Button
          containerStyle={style.signoutButtonContainer}
          buttonStyle={style.signoutButton}
          icon={<Icon name="md-log-out" color="#7d62d9" size={28}></Icon>}
          onPress={() => {
            Alert.alert('Sign out', 'Do you want to sign out?', [
              { text: 'Cancel' },
              {
                text: 'OK',
                onPress: () => {
                  AsyncStorage.removeItem('currentUser', err => {
                    SendBirdSyncManager.getInstance().reset();
                    const sb = SendBird.getInstance();
                    firebase
                      .messaging()
                      .getToken()
                      .then(fcmToken => {
                        if (fcmToken) {
                          if (Platform.OS === 'ios') {
                            sb.unregisterAPNSPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          } else {
                            sb.unregisterGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
                              sb.disconnect(() => {
                                navigation.replace('Signin', {});
                              });
                            });
                          }
                        } else {
                          sb.disconnect(() => {
                            navigation.replace('Signin', {});
                          });
                        }
                      });
                  });
                }
              }
            ]);
          }}
        />
      )
    };
  };
  static getDerivedStateFromProps(props, state) {
    state.channels = props.channels;
    return state;
  }
  constructor(props) {
    super(props);
    this.collection = null;
    this.state = {
      channels: []
    };
  }
  componentDidMount() {
    const { navigation } = this.props;
    const userFromParams = navigation.getParam('user', null);
    if (!userFromParams) {
      AsyncStorage.getItem('currentUser', (err, result) => {
        if (result) {
          const user = JSON.parse(result);
          this.init(user.userId);
        } else {
          this.requireLogin();
        }
      });
    } else {
      this.init(userFromParams.userId);
    }
  }
  componentWillUnmount() {
    if (this.onTokenRefreshListener) {
      this.onTokenRefreshListener();
    }
    if (this.onPushNotificationListener) {
      this.onPushNotificationListener();
    }
    if (this.onPushNotificationOpened) {
      this.onPushNotificationOpened();
    }
    this.props.clearChannels();
    const manager = SendBirdSyncManager.getInstance();
    if (manager) {
      manager.clearCache();
    }
    if (this.collection) {
      this.collection = null;
    }
  }
  init(userId) {
    const sb = SendBird.getInstance();
    sb.connect(userId, (_, err) => {
      if (!err) {
        firebase
          .messaging()
          .getToken()
          .then(fcmToken => {
            if (fcmToken) {
              this.registerPushToken(fcmToken);
            }
          });
      }
    });

    firebase
      .messaging()
      .hasPermission()
      .then(enabled => {
        if (!enabled) {
          firebase
            .messaging()
            .requestPermission()
            .then(() => {
              this.addPushNotificationListener();
            })
            .catch(err => {
              console.log('Failed to get permission', err);
            });
        } else {
          this.addPushNotificationListener();
        }
      });
    this.addPushTokenRefreshListener();

    this.onPushNotificationOpened = firebase.notifications().onNotificationOpened(notificationOpen => {
      firebase.notifications().removeAllDeliveredNotifications();
      if (notificationOpen) {
        const notification = notificationOpen.notification;
        if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
          sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
            if (!err) {
              this.openChat(channel);
            }
          });
        }
      }
    });
    firebase.notifications().removeAllDeliveredNotifications();
    const connectionHandler = new sb.ConnectionHandler();
    connectionHandler.onReconnectFailed = () => {
      sb.reconnect();
    };
    connectionHandler.onReconnectStarted = () => {};
    connectionHandler.onReconnectSucceeded = () => {
      const manager = SendBirdSyncManager.getInstance();
      manager.resumeSync();
    };
    sb.addConnectionHandler('connection handler', connectionHandler);

    const options = new SendBirdSyncManager.Options();
    options.messageCollectionCapacity = 2000;
    options.messageResendPolicy = 'automatic';
    options.failedMessageRetentionDays = 7;
    options.maxFailedMessageCountPerChannel = 50;
    options.automaticMessageResendRetryCount = 4;
    SendBirdSyncManager.setup(userId, options, err => {
      if (!err) {
        firebase
          .notifications()
          .getInitialNotification()
          .then(notificationOpen => {
            if (notificationOpen) {
              const notification = notificationOpen.notification;
              if (notification.data && notification.data.channel && notification.data.channel.channel_url) {
                sb.GroupChannel.getChannel(notification.data.channel.channel_url, (channel, err) => {
                  if (!err) {
                    this.openChat(channel);
                  }
                });
              }
            }
          });

        const query = sb.GroupChannel.createMyGroupChannelListQuery();
        query.limit = 50;
        query.includeEmpty = false;
        query.order = 'latest_last_message';

        this.collection = new SendBirdSyncManager.ChannelCollection(query);
        const collectionHandler = new SendBirdSyncManager.ChannelCollection.CollectionHandler();
        collectionHandler.onChannelEvent = (action, channels) => {
          switch (action) {
            case 'insert': {
              this.props.addChannels(channels);
              break;
            }
            case 'update': {
              this.props.updateChannels(channels);
              break;
            }
            case 'move': {
              this.props.moveChannels(channels);
              break;
            }
            case 'remove': {
              this.props.removeChannels(channels);
              break;
            }
            case 'clear': {
              this.props.clearChannels();
              break;
            }
          }
        };
        this.collection.setCollectionHandler(collectionHandler);
        this.next();
      } else {
        console.log(err);
      }
    });
  }
  addPushTokenRefreshListener() {
    this.onTokenRefreshListener = firebase.messaging().onTokenRefresh(fcmToken => {
      this.registerPushToken(fcmToken);
    });
  }
  addPushNotificationListener() {
    this.onPushNotificationListener = firebase.messaging().onMessage(message => {
      this.handlePushNotification(message);
    });
  }
  handlePushNotification(message) {
    // DO NOTHING FOR FOREGROUND PUSH NOTIFICATION
  }
  registerPushToken(fcmToken) {
    const sb = SendBird.getInstance();
    if (Platform.OS === 'ios') {
      // WARNING! FCM token doesn't work in request to APNs.
      // Use APNs token here instead.
      firebase
        .messaging()
        .ios.getAPNSToken()
        .then(token => {
          sb.registerAPNSPushTokenForCurrentUser(token, (result, err) => {
            if (err) {
              console.log(err);
            }
          });
        })
        .catch(error => {
          console.error(error);
        });
    } else {
      sb.registerGCMPushTokenForCurrentUser(fcmToken, (result, err) => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  requireLogin() {
    this.props.navigation.replace('Signin', {});
  }
  createChannel() {
    this.props.navigation.push('Invite', {});
  }
  openChat(channel) {
    if (navigator.top() !== channel.url) {
      this.props.navigation.push('Chat', { channel });
    }
  }
  next() {
    if (this.collection) {
      this.collection.fetch(err => {
        if (err) {
          console.log(err);
        }
      });
    }
  }
  render() {
    return (
      <View style={style.container}>
        <FlatList
          data={this.state.channels}
          keyExtractor={channel => channel.url}
          onEndReached={this.next.bind(this)}
          renderItem={bundle => {
            const channel = bundle.item;
            return <GroupChannelView channel={channel} onPress={this.openChat.bind(this)}></GroupChannelView>;
          }}
        />
        <ActionButton
          buttonColor="#6e5baa"
          degrees={0}
          shadowStyle={style.actionButtonShadow}
          fixNativeFeedbackRadius={true}
          onPress={this.createChannel.bind(this)}
        >
          <Icon name="md-create" style={style.actionButtonIcon} />
        </ActionButton>
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { main } = previousState;
    return { ...main };
  },
  Action
)(MainController);
