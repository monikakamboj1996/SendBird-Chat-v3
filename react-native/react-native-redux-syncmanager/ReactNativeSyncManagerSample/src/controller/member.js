import React from 'react';
import { View } from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/Ionicons';
import { FlatList } from 'react-native-gesture-handler';

import SendBird from 'sendbird';

import { style } from '../style/member';
import Action from '../action/member';
import UserView from '../view/user';

export class MemberController extends React.Component {
  static navigationOptions = ({ navigation }) => {
    const channel = navigation.getParam('channel', null);
    return {
      title: 'Members',
      headerRight: (
        <View style={{ flexDirection: 'row' }}>
          <Button
            containerStyle={style.headerInviteContainer}
            buttonStyle={style.headerInvite}
            icon={<Icon name="md-add" color="#7d62d9" size={28}></Icon>}
            onPress={() => {
              navigation.push('Invite', { channel });
            }}
          />
        </View>
      )
    };
  };
  constructor(props) {
    super(props);
    this.channel = this.props.navigation.getParam('channel', null);
    this.state = {
      members: []
    };
  }
  static getDerivedStateFromProps(props, state) {
    state.members = props.members;
    return state;
  }
  componentDidMount() {
    const sb = SendBird.getInstance();
    const channelHandler = new sb.ChannelHandler();
    channelHandler.onUserJoined = (channel, user) => {
      this.updateChannel(channel);
    };
    channelHandler.onUserLeft = (channel, user) => {
      this.updateChannel(channel);
    };
    sb.addChannelHandler('member', channelHandler);
    this.props.initMembers(this.channel.members);
  }
  updateChannel(channel) {
    this.channel = channel;
    this.props.updateMembers(this.channel.members);
  }
  render() {
    return (
      <View>
        <FlatList
          data={this.state.members}
          keyExtractor={member => member.userId + ''}
          renderItem={bundle => <UserView user={bundle.item} selectable={false} />}
        />
      </View>
    );
  }
}

export default connect(
  previousState => {
    const { member } = previousState;
    return { ...member };
  },
  Action
)(MemberController);
