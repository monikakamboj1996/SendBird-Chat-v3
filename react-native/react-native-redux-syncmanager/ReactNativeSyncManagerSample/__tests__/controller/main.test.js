import 'react-native';
import React from 'react';
import { MainController } from '../../src/controller/main';
import renderer from 'react-test-renderer';

const navigation = {
  setParams: jest.fn(),
  getParam: jest.fn().mockReturnValue({ userId: 123 }),
  navigate: jest.fn(),
  state: {
    params: {
      channelUrl: 'channelUrl',
      isOpenChannel: true
    }
  },
  dispatch: jest.fn(),
  addListener: jest.fn()
};

describe('controller/main', () => {
  it('renders correctly', () => {
    const comp = <MainController navigation={navigation} />;
    const tree = renderer.create(comp).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // TODO: add test cases
});
