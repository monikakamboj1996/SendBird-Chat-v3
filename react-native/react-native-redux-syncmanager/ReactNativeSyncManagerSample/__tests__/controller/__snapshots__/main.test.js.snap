// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controller/main renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f3f3f3",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={2}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderScrollComponent={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    pointerEvents="box-none"
    style={
      Array [
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "elevation": undefined,
            "justifyContent": "flex-end",
            "zIndex": undefined,
          },
        ],
        undefined,
      ]
    }
  >
    <View
      pointerEvents="none"
      style={
        Object {
          "backgroundColor": "transparent",
          "bottom": 0,
          "elevation": undefined,
          "justifyContent": "flex-end",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": undefined,
        }
      }
    />
    <View
      pointerEvents="box-none"
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "elevation": undefined,
              "justifyContent": "flex-end",
              "zIndex": undefined,
            },
          ],
          Object {
            "alignItems": "flex-end",
          },
          Object {
            "paddingVertical": 30,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginHorizontal": 30,
              "zIndex": undefined,
            },
            Object {
              "elevation": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.35,
              "shadowRadius": 3,
            },
            Object {
              "shadowColor": "#333",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 0.6,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(110, 91, 170, 1)",
                "borderRadius": 28,
                "height": 56,
                "width": 56,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderRadius": 28,
                  "height": 56,
                  "justifyContent": "center",
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                  "width": 56,
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 24,
                    "marginTop": -4,
                  }
                }
              >
                +
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
